src/pages/index.js

```diff
@@ -7,97 +7,211 @@ import Section from "../components/Section.js";
+
   return cardElement.generateCard();
 }
 
-//create a section of cards
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      const cardElement = createCard(data);
-      cardSection.addItem(cardElement);
-    },
+//======================= Render initial cards =====================\\
+const cardListEl = document.querySelector(".cards__list");
+let userID;
+let cardSection;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([user, initialCards]) => {
+    //set UI
+    userInfo.setUserInfo(user.name, user.about);
+    userID = user._id;
+
+    //create a section of cards
+    cardSection = new Section(
+      {
+        items: initialCards,
+        renderer: (data) => {
+          const cardElement = createCard(data);
+          cardSection.addItem(cardElement);
+        },
+      },
+      cardListEl
+    );
+
+    //render initial cards
+    cardSection.renderItems(initialCards);
+  })
+  .catch((err) => console.error(err));
+
+//============================= POPUPS =========================\\
+// previw image popup
+const previewImagePopup = new PopupWithImage("#preview-image-popup");
+previewImagePopup.close();
+
+// delete card popup
+const deleteCardPopup = new PopupConfirm("#delete-card-popup", "Deleting...");
+deleteCardPopup.setEventListeners();
+
+// Edit Profile Avatar popup
+const avatarEditButton = document.querySelector(".profile__image-overlay");
+
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-popup",
+  (inputValues) => {
+    editAvatarPopup.showLoading();
+    api
+      .updateProfilePic(inputValues)
+      .then(() => {
+        profileAvatar.src = inputValues.link;
+        editAvatarPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        editAvatarPopup.hideLoading();
+      });
   },
-  cardListEl
+  "Saving..."
 );
 
-//render initial cards
-cardSection.renderItems(initialCards);
+avatarEditButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
+
+editAvatarPopup.setEventListeners();
 
-//======================= Add a new card =====================\\
+// Add new card popup
 const addCardBtn = document.querySelector(".profile__add-button");
-const addCardPopup = new PopupWithForm("#add-card-popup", (inputData) => {
-  //create a new card
-  const newCard = createCard(inputData);
-  addCardPopup.close();
-  cardSection.addItem(newCard);
-});
+
+const addCardPopup = new PopupWithForm(
+  "#add-card-popup",
+  //handleFormSumit
+  (cardData) => {
+    addCardPopup.showLoading();
+    api
+      .addNewCard(cardData)
+      .then((inputData) => {
+        //create a new card
+        const newCard = createCard(inputData);
+        addCardPopup.close();
+        cardSection.addItem(newCard);
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        addCardPopup.hideLoading();
+      });
+  },
+  "Creating..."
+);
 
 addCardBtn.addEventListener("click", () => {
```

Comment:
Note, that it's needed to disable submit button before opening the popup

Reply:
Needs correction: Note, that it's needed to disable submit button before opening the popup. Use validator class methods

Reply:
Kseniya,

I'm not sure how to do this - the add card form is disabled to submit it before it has validating inputs so not sure how i'm supposed to do this.

Am I creating a new variable for the submit button on the add card form and then adding "toggleSubmitButton" from validation?

Reply:
I noticed that you've added the `reset` listener in validator class, so this issue is resolved. Thanks!